### PR TITLE
Rename apiHost to amoBaseUrl

### LINKED ISSUE #2504: Replace submission-api --api-host cli option with --amo-base-url
As a followup to the original discussion in #2458 about the --api-host option.

@willdurand @Rob--W and I have discussed about it one more time during today's issue triage and the following is a summary of the changes we agreed on:

- rename the option from --api-host to --amo-base-url
- the default value for --amo-base-url will be set to "https://addons.mozilla.org/api/v5"

This option is meant to mainly be used by QA to run QA verifications on the staging or dev AMO servers, and it is reasonable to most extension developers will not need to use it or to customize the url, and so we agreed that we may not need to apply any special check or restrictions to the value got from it (well besides being a valid url that doesn't throw when passed to `new URL(...)`).

### PR DESCRIPTION
**Pull request by eviljeff**
fixes #2504 + makes the base url constractor arg a `URL` in `Client` to be consistent with internal use in the class.

### CODE REVIEW

**Review by rpl — APPROVED** [review]
lgtm 

Thanks @eviljeff!

File: src/util/submit-addon.js [hunk 1]
```diff
@@ -354,9 +354,16 @@ export async function signAddon({
     throw new Error(`error with ${xpiPath}: ${statError}`);
   }
 
+  let baseUrl;
+  try {
+    baseUrl = new URL(amoBaseUrl);
+  } catch (err) {
+    throw new Error(`Invalid amoBaseUrl: ${amoBaseUrl}`);
```
Comment (willdurand) [inline src/util/submit-addon.js:361]:
```suggestion
    throw new Error(`Invalid AMO API base URL: ${amoBaseUrl}`);
```

Comment (willdurand) [inline src/util/submit-addon.js:361]:
```suggestion
    throw new Error(`Invalid AMO API base URL: ${amoBaseUrl}`);
```

Comment (willdurand) [reply src/util/submit-addon.js:361]: Also, I don't see a test case for this.

**Review by willdurand — APPROVED** [review]
r+wc

File: src/util/submit-addon.js (hunk 1 above)
Comment (eviljeff) [reply src/util/submit-addon.js:361]: There wasn't one before 😉 but 👍

File: tests/unit/test-util/test.submit-addon.js [hunk 2]
```diff
@@ -150,6 +150,15 @@ describe('util.submit-addon', () => {
         'Error: ENOENT: no such file or directory'
       );
     });
+
+    it('throws error if xpiPath is invalid', async () => {
```
Comment (willdurand) [inline tests/unit/test-util/test.submit-addon.js:154]:
```suggestion
    it('throws error if amoBaseUrl is invalid', async () => {
```

**Pull request merged**